src/xlsx/functions/table.ts

```diff
@@ -42,15 +42,13 @@ function addAutoFilter(table: ExcelFilterTableData): XMLString {
 }
 
 function addFilterColumns(table: ExcelFilterTableData): XMLString[] {
-  const tableZone = toZone(table.range);
   const columns: XMLString[] = [];
-  for (const i of range(0, zoneToDimension(tableZone).width)) {
-    const filter = table.filters[i];
-    if (!filter || !filter.filteredValues.length) {
+  for (const filter of table.filters) {
+    if (!filter || (!filter.filteredValues.length && !filter.displayBlanks)) {
```

Comment:
seems already broken in 16.0, but it should be the other way around I think.
If you uncheck every value in o-spreadsheet and export, all values are checked.

Reply:
The condition was indeed wrong, but it turns out that the main problem is that we're not supposed to filter every value in the table column. It's impossible to do that in Excel and Calc. And it's a bit buggy if we import a filter where everything is filtered.

We could :
1) Disable filters where everything is filtered at export. That's really annoying 'cause we'd need to recompute teh hidden rows taking those disabled filters into account.
2) Don't allow a filter to filter everything. That's actually really annoying too, because our filters are dynamic (filtered row change when cell values change), which is not the case in Excel. We'd have to handle the case where a cell value change make every value filtered.
3) Accept that the export is a bit buggy in this case.

Reply:
as discussed:
there's no easy fix, and the issue is minor: if you change anything in the filter in excel: everything is back to normal
so 3. seems reasonable